tests/vmi_multus_test.go

```diff
@@ -864,29 +864,43 @@ var _ = Describe("[Serial]SRIOV", func() {
 			waitVmi(vmi1)
 			waitVmi(vmi2)
 
+			waitForVMIGuestAgentReport(virtClient, vmi1.Namespace, vmi1.Name, 2)
```

Comment:
consider calling this from waitVmi, so that it will be aplied for all SRIOV tests

Reply:
we dont need to, because as Edy said, once Quique will fix #4432 we wont need this on any other than the two tests i am using now and 3985
(telling according the flakes i see that use the status, and need the guest agent report)

waitVmi is used all over the tests, it shouldnt do unneeded stuff that takes time imo.

Reply:
i will add waitForVMIGuestAgentReport meanwhile to 3985, i wanted to leave it for other PR as we discussed offline, but since you dont mind that i will add it and even ask in this PR + it would help me to test without failures, i found it suitable.

Reply:
> we dont need to, because as Edy said, once Quique will fix #4432 we wont need this on any other than the two tests i am using now and 3985
> (telling according the flakes i see that use the status, and need the guest agent report)
> 
> waitVmi is used all over the tests, it shouldnt do unneeded stuff that takes time imo.

OK. Understand. Thanks for explaining.
> 
> Edit
> As i wrote yesterday in the PR description, i intentionally dont mess with this test 3985 in this PR, this PR should be focused, we shouldnt fix that many distinct problems in one PR, it affects the history and the flow, we can simply create another PR once we are fine with this one, as for now, its not even lgtm approve yet, once it will be we can use the function waitForVMIGuestAgentReport safely on 3985 either on this PR or a follow PR, but we should converge first.

> i will add waitForVMIGuestAgentReport meanwhile to 3985, i wanted to leave it for other PR as we discussed offline, but since you dont mind that i will add it and even ask in this PR + it would help me to test without failures, i found it suitable.  

Then isn't the best solution will be to make the test fix, in a different PR, and rebase this PR on top of it?
It will keep the flow and history correct, and test lane will be functional. we don't know how long https://github.com/kubevirt/kubevirt/pull/4335/commits/200777a0105b422dcdfb05b7afed8442b43c377a is going to take, and in the meanwhile we are loosing time with skipped and flaky tests.

Reply:
i can do that, but please keep this in mind,
https://github.com/kubevirt/kubevirt/pull/4335#issuecomment-720573589

it would still not be perfect due to other bug of missing reports,
lets see if its crucial according @EdDev that i will split and move  waitForVMIGuestAgentReport other PR

Reply:
OK. Let's wait then.

Reply:
Done

Reply:
@alonSadan 
lets continue https://github.com/kubevirt/kubevirt/pull/4335#issuecomment-721124676 here
(marked as unresolved again)

It doesnt flake on CI so far, just locally, and the tests of it are skipped on CI, i dont see any reason to split the PRs.
in case you will need it for VLAN or any test, let me know 1/2 day before and i will split it, its it fine ?

You can either rebase on this as well, and you will need also Quique PR which isnt merged.

Reply:
Let me do a quick research soon, it might be easy to split it, if its that important for you.

Reply:
1. test 3985: is not flaky on CI but locally it does.  your waiting function should help with that.

2. tests 3956,3957: flake locally and on CI. 
your waiting function should help with that as well?

Reply:
2. they are skipped on CI, because the IP issue Quique fixes,
and with my addon i need the waitFor function, because i check the name there, without it it would be flaky.

Note having said all this, dont forget please the hiccups issue,
i would suggest, lets focus if you need it or CI, not if locally i need it ?
this PR would make it more stable, but what should affect if we want to split is either you need it or CI need it,
for the first you can rebase, but if its that important for you i will split.

Reply:
> I would suggest, lets focus if you need it or CI, not if locally i need it ?
this PR would make it more stable, but what should affect if we want to split is either you need it or CI need it,
for the first you can rebase, but if its that important for you i will split (if you need to use the code).

OK. so  from side, keep it here , and split it in case it will be needed.

Reply:
sounds good, but it would still be only possible after i talk with Edy on the (*)
as the (*) adds flakiness, because its a real bug, we can workaround it, all info at the original thread about it.

Reply:
https://prow.apps.ovirt.org/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/4463/pull-kubevirt-e2e-kind-1.17-sriov/1323866556106543104

now we see 3985 flakes on CI as well,
we didnt see it before just because all was red due to #4470